src/response.rs

```diff
@@ -161,85 +128,15 @@ impl Response {
     /// This sets the response `Content-Type` header.
     ///
     /// [Read more on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types)
-    #[must_use]
-    pub fn set_content_type(mut self, mime: impl Into<Mime>) -> Self {
+    pub fn set_content_type(&mut self, mime: impl Into<Mime>) {
         self.res.set_content_type(mime.into());
-        self
-    }
-
-    /// Pass a string as the response body.
-    ///
-    /// # Mime
-    ///
-    /// The content type is set to `text/plain; charset=utf-8`.
-    #[must_use]
-    pub fn body_string(mut self, string: String) -> Self {
```

Comment:
Not so sure about removing these - are we depending on always mime sniffing now? There's definitely some overhead if that's the case.

Otherwise the docs should probably be clear that the user needs to set it themselves.

Reply:
> Not so sure about removing these - are we depending on always mime sniffing now? There's definitely some overhead if that's the case.

These methods have been moved to rely on [`Body`](https://docs.rs/http-types/2.1.0/http_types/struct.Body.html) instead and go through the `set_body` method instead:

```rust
// old API
res.body(io::Cursor::new(b"hello world".to_owned()));
res.body_string("hello world".to_string());

// new API
res.set_body(b"hello world");
res.set_body("hello world");
```

The only set of APIs that become a bit more awkward by this are the Serde-related APIs. But at least https://github.com/http-rs/tide/pull/523 is showing us improvements for some applications, and with a `impl From<Body> for Response` bound we could move this even further ahead (in a future patch):

```rust
// old API
res.body_json(Cat { name: "chashu" });

// new API
res.set_body(Body::from_json(Cat { name: "chashu" });

// http-rs/tide#523 (return type from function)
Ok(json!({ "name": "chashu" }));

// future option #1 
res.set_body(json!({ "name": "chashu" });

// future option #2 (return type from function)
Ok(Body::from_json(Cat { name: "chashu" }));
```

Either way, I hope this shows what the intended workflow around Response bodies is with this patch, and shows some options we could explore in the future to improve this further.